src/main/java/seedu/toluist/controller/FindController.java

```diff
@@ -0,0 +1,163 @@
+package seedu.toluist.controller;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.logging.Logger;
+
+import seedu.toluist.commons.core.LogsCenter;
+import seedu.toluist.dispatcher.CommandResult;
+import seedu.toluist.model.Task;
+import seedu.toluist.model.TodoList;
+import seedu.toluist.ui.Ui;
+
+/**
+ * Searches the task list for matches in the parameters, and displays the results received
+ */
+public class FindController extends Controller {
+    private static final String COMMAND_TEMPLATE = "^(?<command>(find|filter|list))"
+            + "(\\s+(?<keywords>.+))?\\s*";
+    private static final String SEARCH_BY_TAG = "tag/";
```

Comment:
variable name should reflectt what it is, generally variable/class names should be nouns while method names should be verbs